mpmp/src/model/Player.java

```diff
@@ -323,40 +299,21 @@ public static Color parseColor(String s) {
 	}
 
 	public static int numPlayers() {
-		return players.size();
-	}
-
-	public static Player getCurrentPlayer() {
-		return currentPlayer;
-	}
-
-	public static void setCurrentPlayer(Player p) {
-		currentPlayer = p;
+		return subscribed.size();
 	}
 
 	public static String matches(String name, boolean at) {
-		Iterator i = players.iterator();
-		String bestMatch = "";
-		while (i.hasNext()) {
-			Player p = (Player) i.next();
-			String pname = p.getName();
-			if (at) {
-				if (name.matches("@" + pname + "(.*)") && pname.length() > bestMatch.length()) {
-					bestMatch = pname;
-				}
-			} else if (name.matches(pname + "(.*)") && pname.length() > bestMatch.length()) {
-				bestMatch = pname;
-			}
+		String best = "";
 
-		}
-		if (bestMatch.equals("")) {
-			return null;
-		}
-		return bestMatch;
+		for (Player p : subscribed)
+			if (name.matches(((at)?"@":"") + p.name + "(.*)") && p.name.length() > best.length())
```

Comment:
The `(.*)` part is always redundant, but I left it in for no reason :confused: All it does is match zero or more characters of any type.